### ISSUE TITLE
Serializing an array?

### ISSUE THREAD

**Issue by mattio** [openmhealth/Granola#2]
It's clear how to serialize a single sample, but is there a way to serialize an array of results easily? 

I have an array of HealthKit steps that I'm querying for and I'd like to serialize them to JSON.

**Comment by schaefba** [comment]
Thanks for the question. Granola currently has no methods or classes that operate on arrays of HealthKit samples. Though, I want to make sure I am understanding your needs correctly. Could you discuss your use case a bit more? Are you interested in serializing an array of HealthKit samples into a JSON array containing serialized representations of those samples? Or do you have a different use case in mind?

If you are referring to the case of serializing an array of samples into a JSON array, as it stands, you would need to: 1) iterate through an array of HealthKit samples, 2) serialize each sample, 3) accumulate those serialized samples into an array, and then 4) serialize that array using the NSJSONSerialization class.

**Comment by mattio** [comment]
What you outlined is correct -- sorry I didn't make that clearer. I found Granola while searching for options to serialize HealthKit data and send it to a server.

Ideally what we're targeting is a way to get the samples for an entire day serialized and submitted to our servers. Nothing too complicated at all at this point. We'd like to retain the detail on sample objects, which is why we're not summarizing counts for the entire day in a statistics query.

What you outlined is pretty much how I handled it so far. I wanted to make sure I wasn't missing an obvious alternative.
